File: app/src/main/java/com/e/myfavoritemovies/Database/FavoriteMovieEntry.java
package com.e.myfavoritemovies.Database;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Ignore;
import android.arch.persistence.room.PrimaryKey;

@Entity(tableName="FavoriteMovie")
public class FavoriteMovieEntry {

    @PrimaryKey(autoGenerate = true)
    private int id;

    private String movieId;
    private String title;

    public FavoriteMovieEntry(int id, String title, String movieId){
        this.id=id;
        this.title=title;
        this.movieId=movieId;
    }

    @Ignore
    public FavoriteMovieEntry(String title,String movieId){
        this.title=title;
        this.movieId=movieId;
    }

    public String getTitle() {
        return title;
    }
    
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getMovieId() {
        return movieId;
    }

}

File: app/src/main/java/com/e/myfavoritemovies/Database/FavoriteMovieEntry.kt
package com.e.myfavoritemovies.Database

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey

@Entity(tableName = "FavoriteMovie")
data class FavoriteMovieEntry(
    @PrimaryKey(autoGenerate = true)
    var id: Int,
    val title: String,
    val movieId: String
) {
    @Ignore
    constructor(title: String, movieId: String) : this(0, title, movieId)
}
